### PR TITLE
Add ControlPlaneRequestEncoder

### PR DESCRIPTION
**Pull request by fabianfett**
Add a new `ControlPlaneRequestEncoder` that encodes all requests into an existing buffer. 

### Motivation:

We want to write our request to the ControlPlaneAPI into one buffer, that we can reuse over and over. 

### Modifications:

- Add `ControlPlaneRequestEncoder` that writes `ControlPlaneRequest`s into a buffer that is reused.

### Result:

- Fewer messages on the ChannelPipeline.

### CODE REVIEW

File: Sources/AWSLambdaRuntimeCore/ControlPlaneRequestEncoder.swift [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+            context.flush()
+
+        case .invocationResponse(let requestID, let payload):
+            let contentLength = payload?.readableBytes ?? 0
+            self.byteBuffer.writeInvocationResultRequestLine(requestID)
+            self.byteBuffer.writeHostHeader(host: self.host)
+            self.byteBuffer.writeString(.userAgentHeader)
+            self.byteBuffer.writeContentLengthHeader(length: contentLength)
+            self.byteBuffer.writeString(.CRLF) // end of head
+            if contentLength > 0 {
```
Comment (tomerd) [inline Sources/AWSLambdaRuntimeCore/ControlPlaneRequestEncoder.swift:46]: `if let payload = payload && contentLength > 0` would be nice so we can drop the force unwrap

File: Sources/AWSLambdaRuntimeCore/ControlPlaneRequestEncoder.swift [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+            self.byteBuffer.writeString(.CRLF) // end of head
+            self.byteBuffer.writeBytes(payload)
+            context.write(self.wrapOutboundOut(self.byteBuffer), promise: promise)
+            context.flush()
+        }
+    }
+
+    mutating func writerAdded(context: ChannelHandlerContext) {
+        self.byteBuffer = context.channel.allocator.buffer(capacity: 256)
+    }
```
Comment (tomerd) [inline Sources/AWSLambdaRuntimeCore/ControlPlaneRequestEncoder.swift:82]: nice

**Review by tomerd — APPROVED** [review]
😎

File: Sources/AWSLambdaRuntimeCore/ControlPlaneRequestEncoder.swift (hunk 1 above)
Comment (fabianfett) [reply Sources/AWSLambdaRuntimeCore/ControlPlaneRequestEncoder.swift]: Fixed.

**Pull request merged**